lib/logstash/outputs/stomp.rb

```diff
@@ -57,11 +66,35 @@ def register
     
     connect
   end # def register
-  
-  def receive(event)
-      
 
-      @logger.debug(["stomp sending event", { :host => @host, :event => event }])
-      @client.send(event.sprintf(@destination), event.to_json)
+  public
+  def do_close
+    @logger.warn("Disconnecting from stomp broker")
+    @client.disconnect if @client.connected?
+  end # def do_close
+
+  def multi_receive(events)
+
+    headers = Hash.new
+    if @headers
+      @headers.each do |k,v|
+        headers[k] = event.sprintf(v)
+      end
+    end
+
+    if @batch
+      @logger.debug(["stomp sending events in batch", { :host => @host, :events => events.length, :headers => headers }])
+
+      @client.transaction do |t|
+        events.each { |event|
+          t.send(event.sprintf(@destination), event.to_json, headers)
+        }
+      end
+    else
+      events.each { |event|
+        @logger.debug(["stomp sending event", { :host => @host, :event => event, :headers => headers }])
+        @client.send(event.sprintf(@destination), event.to_json, headers)
+      }
+    end
```

Comment:
Ok, I'm removing the batch support then. Nothing needs to be enabled on the broker side. It works out of the box.